ludwig/api.py

```diff
@@ -914,6 +914,48 @@ def _tune_batch_size(self, trainer, dataset, random_seed: int = default_random_s
         trainer.eval_batch_size = self.config_obj.trainer.eval_batch_size
         trainer.gradient_accumulation_steps = self.config_obj.trainer.gradient_accumulation_steps
 
+    def save_upscaled_quantized_model(self, save_path: str) -> None:
+        """Upscales quantized weights of a model to fp16 and saves the result in a specified folder.
+
+        Args:
+            save_path (str): The path to the folder where the upscaled model weights will be saved.
+
+        Raises:
+            ValueError:
+                If the model type is not 'llm' or if quantization is not enabled or the number of bits is not 4 or 8.
+            RuntimeError:
+                If no GPU is available, as GPU is required for quantized models.
+
+        Returns:
+            None
+        """
+        if self.config_obj.model_type != MODEL_LLM:
+            raise ValueError(
+                f"Model type {self.config_obj.model_type} is not supported by this method. Only `llm` model type is "
+                "supported."
+            )
+
+        if not self.config_obj.quantization:
+            raise ValueError(
+                "Quantization is not enabled in your Ludwig model config. "
+                "To enable quantization, set `quantization` to `{'bits': 4}` or `{'bits': 8}` in your model config."
+            )
+
+        if self.config_obj.quantization.bits != 4:
+            raise ValueError(
+                "This method only works with quantized models with 4 bits. "
+                "Support for 8-bit quantized models will be added in a future release."
+            )
+
+        if not torch.cuda.is_available():
+            raise RuntimeError("GPU is required for quantized models but no GPU found.")
```

Comment:
If this ever becomes a common set of checks, it could be worth creating a reusable function.

Reply:
Agreed!